go/vt/schemadiff/table.go

```diff
@@ -2197,20 +2250,19 @@ func (c *CreateTableEntity) apply(diff *AlterTableEntityDiff) error {
 				return &ApplyKeyNotFoundError{Table: c.Name(), Key: opt.Name.String()}
 			}
 
-			// Now, if this is a normal key being dropped, let's validate it does not leave any foreign key constraint uncovered
-			switch opt.Type {
-			case sqlparser.PrimaryKeyType, sqlparser.NormalKeyType:
-				for _, cs := range c.CreateTable.TableSpec.Constraints {
-					fk, ok := cs.Details.(*sqlparser.ForeignKeyDefinition)
-					if !ok {
-						continue
-					}
-					if !c.columnsCoveredByInOrderIndex(fk.Source) {
-						return &IndexNeededByForeignKeyError{Table: c.Name(), Key: opt.Name.String()}
-					}
+		case *sqlparser.RenameIndex:
+			// validate no existing key by same name
+			newKeyName := opt.NewName.String()
+			for _, index := range c.TableSpec.Indexes {
+				if strings.EqualFold(index.Info.Name.String(), newKeyName) {
+					return &ApplyDuplicateKeyError{Table: c.Name(), Key: newKeyName}
+				}
+			}
+			for _, index := range c.TableSpec.Indexes {
+				if index.Info.Name.String() == opt.OldName.String() {
+					index.Info.Name = opt.NewName
 				}
 			}
```

Comment:
Do we need to error here if the old name doesn't exist?

Reply:
Yes, done!